spec/observables/defer-spec.ts

```diff
@@ -72,4 +72,16 @@ describe('Observable.defer', () => {
     expectObservable(e1, unsub).toBe(expected);
     expectSubscriptions(source.subscriptions).toBe(sourceSubs);
   });
+
+  type(() => {
```

Comment:
feasible opinion. In most cases test cases already consumes type definitions, so probably won't need to be covered once again via explicit. I remember few cases that test case does not include all of possible type definition cases though - start with opt out and only add necessary cases once it's being noticed maybe better way to go?